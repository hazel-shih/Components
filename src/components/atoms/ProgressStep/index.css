:root {
	--line-width: var(--line-width);
}

.progress-step {
	width: 50%;

	& > .progress-container {
		display: flex;
		justify-content: space-between;
		margin-bottom: 25px;
		position: relative;

		& > .line {
			position: absolute;
			background-color: var(--primary);
			height: 2px;
			max-width: 100%;
			width: var(--line-width);
			z-index: -1;
			top: 50%;
			transform: translateY(-50%);
			transition: 0.4s ease;
		}

		& > .progress {
			display: flex;
			justify-content: center;
			align-items: center;
			background-color: white;
			width: 30px;
			height: 30px;
			border-radius: 50%;
			color: var(--neutral-n3-info-text);
			font-weight: 500;
			font-size: 12px;
			transition: 0.4s ease;
			border: 2px solid var(--neutral-n5-div);

			&.active {
				border: 2px solid var(--primary);
			}
		}
	}

	& > .progress-container::before {
		content: '';
		position: absolute;
		background-color: var(--neutral-n5-div);
		height: 2px;
		width: 100%;
		z-index: -1;
		top: 50%;
		transform: translateY(-50%);
	}

	& > .buttons {
		display: flex;
		justify-content: center;

		& > button {
			background-color: var(--primary);
			border-radius: 8px;
			padding: 8px 30px;
			color: white;
			font-weight: 500;

			& + button {
				margin-left: 50px;
			}

			&:active {
				transform: scale(0.98);
			}

			&:disabled {
				background-color: var(--neutral-n5-div);
				cursor: not-allowed;

				&:active {
					transform: unset;
				}
			}
		}
	}
}
