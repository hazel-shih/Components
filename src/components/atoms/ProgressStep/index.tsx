/* eslint-disable react/no-array-index-key */
import React, { useState, CSSProperties } from 'react';
import classnames from 'classnames';

import styles from './index.css';

interface ProgressStepProperty {
	progressNum: number;
	className?: string;
}

const ProgressStep: React.FC<ProgressStepProperty> = ({ progressNum, className }) => {
	// start from 0
	const [currentProgress, setCurrentProgress] = useState<number>(0);
	return (
		<div className={classnames(styles.progressStep, className)}>
			<div className={styles.progressContainer}>
				<div
					className={styles.line}
					style={
						{ '--line-width': `${(currentProgress / (progressNum - 1)) * 100}%` } as CSSProperties
					}
				/>
				{Array(progressNum)
					.fill(null)
					.map((_, index) => (
						<div
							className={classnames(styles.progress, {
								[styles.active]: index <= currentProgress,
							})}
							key={index}
						>
							{index + 1}
						</div>
					))}
			</div>
			<div className={styles.buttons}>
				<button
					type="button"
					onClick={() => {
						setCurrentProgress(currentProgress - 1);
					}}
					disabled={currentProgress === 0}
				>
					Prev
				</button>
				<button
					type="button"
					onClick={() => {
						setCurrentProgress(currentProgress + 1);
					}}
					disabled={currentProgress === progressNum - 1}
				>
					Next
				</button>
			</div>
		</div>
	);
};

export default ProgressStep;
