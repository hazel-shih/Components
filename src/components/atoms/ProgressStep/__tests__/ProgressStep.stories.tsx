import React from 'react';
import { storiesOf } from '@storybook/react';

import ProgressStep from 'components/atoms/ProgressStep';

const stories = storiesOf('atoms/ProgressStep', module);

stories.add('__interactive', () => <ProgressStep progressNum={5} />);
