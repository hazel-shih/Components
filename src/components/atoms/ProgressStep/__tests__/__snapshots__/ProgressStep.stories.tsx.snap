// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="progressStep"
  >
    <div
      class="progressContainer"
    >
      <div
        class="line"
        style="--line-width: 0%;"
      />
      <div
        class="progress active"
      >
        1
      </div>
      <div
        class="progress"
      >
        2
      </div>
      <div
        class="progress"
      >
        3
      </div>
      <div
        class="progress"
      >
        4
      </div>
      <div
        class="progress"
      >
        5
      </div>
    </div>
    <div
      class="buttons"
    >
      <button
        disabled=""
        type="button"
      >
        Prev
      </button>
      <button
        type="button"
      >
        Next
      </button>
    </div>
  </div>
</DocumentFragment>
`;
